packages/dataviews/src/dataviews-layouts/grid/index.tsx

```diff
@@ -71,14 +72,15 @@ function GridItem< Item >( {
 	regularFields,
 	badgeFields,
 	hasBulkActions,
+	sizes,
 }: GridItemProps< Item > ) {
 	const { showTitle = true, showMedia = true, showDescription = true } = view;
 	const hasBulkAction = useHasAPossibleBulkAction( actions, item );
 	const id = getItemId( item );
 	const instanceId = useInstanceId( GridItem );
 	const isSelected = selection.includes( id );
 	const renderedMediaField = mediaField?.render ? (
-		<mediaField.render item={ item } />
+		<mediaField.render item={ item } sizes={ sizes } />
```

Comment:
Thanks for looking into this.

One advantage of using this approach instead of passing the view object is that we can use information that may be private to the view to calculate the size.

One downside is that we need to figure out is how to support any kind of media field (or the most common ones). That I know of, I think we've used images and iframes as media fields. Perhaps video would be useful as well, though I'm not convinced. We'd need to understand what each of them needs, so we can support them.

To discriminate among the different types of media fields, we need to evolve the field types available. For example, introducing image, iframe, etc. field types. That could signal which props to use.

Reply:
Yeah, I think the idea of making it more generic is a good one and it'll be more flexible. Maybe it could be called `mediaAppearance`.

I also personally prefer more specific types like `VideoAppearance` / `ImageAppearance` / `BlockPreviewAppearance`, as then the user can't specify an invalid combination of props, but this discussion has also come up on other PRs and there are mixed opinions about it.

Reply:
Yes, that could work!